core/services/vrf/listener_v2.go

```diff
@@ -46,13 +49,72 @@ const (
 		6685 // Positive static costs of argument encoding etc. note that it varies by +/- x*12 for every x bytes of non-zero data in the proof.
 )
 
+func newListenerV2(
+	cfg Config,
+	l logger.Logger,
+	ethClient evmclient.Client,
+	chainID *big.Int,
+	logBroadcaster log.Broadcaster,
+	q pg.Q,
+	coordinator vrf_coordinator_v2.VRFCoordinatorV2Interface,
+	aggregator *aggregator_v3_interface.AggregatorV3Interface,
+	txm txmgr.TxManager,
+	pipelineRunner pipeline.Runner,
+	gethks keystore.Eth,
+	job job.Job,
+	reqLogs *utils.Mailbox,
+	reqAdded func(),
+	respCount map[string]uint64,
+	headBroadcaster httypes.HeadBroadcasterRegistry,
+	deduper *logDeduper,
+) (*listenerV2, error) {
+	coordinatorABI, err := abi.JSON(strings.NewReader(vrf_coordinator_v2.VRFCoordinatorV2ABI))
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	return &listenerV2{
+		cfg:                cfg,
+		l:                  l,
+		ethClient:          ethClient,
+		chainID:            chainID,
+		logBroadcaster:     logBroadcaster,
+		txm:                txm,
+		coordinator:        coordinator,
+		coordinatorABI:     coordinatorABI,
+		pipelineRunner:     pipelineRunner,
+		job:                job,
+		q:                  q,
+		gethks:             gethks,
+		reqLogs:            reqLogs,
+		chStop:             make(chan struct{}),
+		reqAdded:           reqAdded,
+		respCount:          respCount,
+		blockNumberToReqID: pairing.New(),
+		headBroadcaster:    headBroadcaster,
+		latestHeadMu:       sync.RWMutex{},
+		wg:                 &sync.WaitGroup{},
+		aggregator:         aggregator,
+		deduper:            deduper,
+	}, nil
+}
+
 type pendingRequest struct {
 	confirmedAtBlock uint64
 	req              *vrf_coordinator_v2.VRFCoordinatorV2RandomWordsRequested
 	lb               log.Broadcast
 	utcTimestamp     time.Time
 }
 
+type vrfPipelineResult struct {
+	err         error
```

Comment:
Don't feel strongly about this, but one pro of putting err outside the result struct rather and as one of the return values is that the caller is compelled (by way of linting / go vet / etc.) to either completely ignore the err by underscoring or to check it, I think this way might hide it a bit.

Reply:
The reason I did it this way is that otherwise we don't know which err is associated with which pipeline, so we don't know which one to skip, which reqID label to set when logging the error etc